torch_xla/csrc/runtime/distributed_runtime.h

```diff
@@ -3,33 +3,78 @@
 
 #include <memory>
 
+#include "torch_xla/csrc/runtime/debug_macros.h"
+#include "tsl/distributed_runtime/preemption/preemption_sync_manager.h"
 #include "xla/pjrt/distributed/distributed.h"
 
 namespace torch_xla {
 namespace runtime {
 
+// DistributedRuntime serves as the point of entry for all operations which
+// required the XLA distributed runtime, such as preemption coordination.
 class DistributedRuntime {
  public:
-  static const std::string default_coordinator_port;
-  static DistributedRuntime& getInstance(int global_rank,
-                                         std::string master_addr,
-                                         std::string port) {
-    static DistributedRuntime dist_runtime_instance(global_rank, master_addr,
-                                                    port);
-    return dist_runtime_instance;
+  static inline const std::string kDefaultCoordinatorPort = "8547";
+
+  // Returns true if the distributed runtime has already been initialized.
+  static bool IsInitialized() { return dist_runtime_ != nullptr; }
+
+  // Initialize the shared DistributedRuntime object. This creates a
+  // DistributedRuntimeClient on each worker, and on global_rank 0 initializes
+  // the corresponding DistributedRuntimeService.
+  static void Initialize(int global_rank, int world_size,
+                         std::string master_addr, std::string port) {
+    XLA_CHECK(!IsInitialized()) << "DistributedRuntime already initialized";
+    dist_runtime_ = std::unique_ptr<DistributedRuntime>(
```

Comment:
Would it guarantee the DistributedRuntime ctor only run once?

Reply:
The `unique_ptr` takes ownership of the raw pointer, so the constructor is only called once. Due to the visibility of the private constructor, we can't directly use `make_unique`.